AQ3DC/AQ3DC.py

```diff
@@ -0,0 +1,1062 @@
+  dic_patient = {} 
+  dic_time = {}
+  dic_tooth = {}
+  normpath_T1 = os.path.normpath("/".join([landmarks_dir_T1,'**','']))
+  normpath_T2 = os.path.normpath("/".join([landmarks_dir_T2,'**','']))
+  lst_normpath = [normpath_T1,normpath_T2]
+  
+  for normpath in lst_normpath:
+    for jsonfile in sorted(glob.iglob(normpath, recursive=True)):
+      # print(jsonfile)
+      if os.path.isfile(jsonfile) and True in [ext in jsonfile for ext in [".json"]]:
+        num_patient= os.path.basename(jsonfile).split("_")[2][1:]
+        # print(num_patient)
+        time = os.path.basename(jsonfile).split("_")[1]
+        if "_L." in jsonfile:
+          if num_patient in dic_patient.keys():
+            if time in dic_time.keys():
+              dic_time[time]['path_landmark_L'] = jsonfile
+              dic_patient[num_patient] = dic_time
+            else :
+              dic_time[time] = {'path_landmark_L' : jsonfile}
+              dic_patient[num_patient] = dic_time
+          else:
+            dic_time[time] = {'path_landmark_L' : jsonfile}
+            dic_patient[num_patient] = dic_time
+        else :
+          if num_patient in dic_patient.keys():
+            if time in dic_time.keys():
+              dic_time[time]['path_landmark_U'] = jsonfile
+              dic_patient[num_patient] = dic_time
+            else :
+              dic_time[time] = {'path_landmark_U' : jsonfile}
+              dic_patient[num_patient] = dic_time
+          else:
+            dic_time[time] = {'path_landmark_U' : jsonfile}
+            dic_patient[num_patient] = dic_time
+  
+  # print(dic_patient)\
+  list_label = []
+  for obj in dic_patient.items():
+    obj=obj[1]    
+    for patient_t in obj.items():
+      time = patient_t[0]
+      dic_path_patient = patient_t[1]
+      # print(dic_path_patient)
+      # print(dic_path_patient.items())
+      for path_patient in dic_path_patient.items():
+        # print('path_patient :',path_patient)
+        json_file = pd.read_json(path_patient[1])
+        markups = json_file.loc[0,'markups']
+        controlPoints = markups['controlPoints']
+        for i in range(len(controlPoints)):
+          label = controlPoints[i]["label"]#.split("-")[1]
+          if label not in list_label:
+            list_label.append(label)
+          tooth = label[:3]
+          # print(tooth)
+          type_land = label[3:]
+          position = controlPoints[i]["position"]
+          if tooth not in dic_tooth.keys():
+            dic_tooth[tooth] = {}
+          if type_land not in dic_tooth[tooth]:
+            dic_tooth[tooth][type_land] = {}
+          
+          dic_tooth[tooth][type_land][time] = position
+  # print(list_label)
+  return dic_tooth
+
+DistanceResult = collections.namedtuple("DistanceResult", ("delta", "norm"))
+def computeDistance(point1, point2) -> DistanceResult:
+    delta = np.abs(np.subtract(point2,point1))
+    return DistanceResult(
+        delta,
+        np.linalg.norm(delta),
+    )
```

Comment:
This can be imported from the `DependantMarkups` module.

```python3
from DependantMarkups import DependantMarkupsLogic
```

You should then have an instance of this as an attribute in `AQ3DCLogic`:

```python3
  def __init__(self):
    self.dependencies = DependantMarkupsLogic()
```

You can then use methods like `self.dependencies.computeDistance` or `self.dependencies.setProjected` as described in [/DependantMarkups/README.md](https://github.com/DCBIA-OrthoLab/Q3DCExtension/blob/dependant-markups/DependantMarkups/README.md)